catchup/service.go

```diff
@@ -464,3 +491,55 @@ func (s *Service) sync() {
 
 	s.log.Infof("Catchup Service: finished catching up, now at round %v (previously %v). Total time catching up %v.", s.ledger.LastRound(), pr, elapsedTime)
 }
+
+// nextRoundIsNotApproved returns true if the next round upgrades to a protocol version
+// which is not approved.
+// In case of an error, it returns false
+func (s *Service) nextRoundIsNotApproved(nextRound basics.Round) bool {
+	lastLedgerRound := s.ledger.LastRound()
+	proto, err := s.ledger.ConsensusParams(lastLedgerRound)
+	if err != nil {
+		s.log.Errorf("catchup: could not get consensus parameters for round %v: $%v", lastLedgerRound, err)
+		s.log.Errorf("catchup: could not determine if next round is from an approved protocol.")
```

Comment:
I don't think that this error logging is needed. Generating multiple errors for the same issue isn't very helpful, and creates redundant verbosity.

Reply:
Will drop the second error log.